Configuring a custom domain on AWS (DOC-16) (#27)
openops-cloud/docs: getting-started/deployment/aws-ec2.mdx

sudo docker compose down && sudo docker compose up -d
```

## Enabling TLS (HTTPS) access with a load balancer

> For production usage, it's recommended to enable TLS (HTTPS). Without TLS, cloud templates may not load properly when using the Safari browser.
> 

Reviewer: ```suggestion
For production usage, it's recommended to enable TLS (HTTPS). In addition to the security aspect, this will also ensure cloud templates load properly on different browsers.
```

Author: Incorporated the suggestion

To secure OpenOps with TLS, it's recommended to place your EC2 instance behind an AWS Application Load Balancer (ALB). This setup uses AWS Certificate Manager (ACM) to handle TLS, and allows secure access via a DNS name, even without a [custom domain](#accessing-openops-via-a-custom-domain-over-tls).

<Warning>Once the ALB is in place, all access should go through it — you can no longer use the EC2 public IP directly.</Warning>

### Request a TLS certificate in ACM

1. Go to the [AWS Certificate Manager](https://console.aws.amazon.com/acm/home) (ACM).

> 2. Request a public certificate for your EC2's public DNS name (e.g., `ec2-1-2-3-4.compute-1.amazonaws.com`).

Reviewer: that won't work, you need a custom domain.

https://stackoverflow.com/a/66687070/24956286

    * **Protocol**: HTTP
    * **Port**: 80
    * **Health check path**: `/api/v1/health`
5. Register your EC2 instance as a target.
6. Finish creating the ALB and wait for its status to become **Active**.

Once setup is complete, you'll be able to access OpenOps securely at `https://your-alb-dns-name.elb.amazonaws.com`.

### Update environment and restart

1. SSH into your EC2 instance or use CloudShell.
2. Update the public OpenOps URL in the `.env` file:
    ```env
    OPS_PUBLIC_URL=https://your-alb-dns-name.elb.amazonaws.com
    ```
3. Restart the containers:
    ```bash
    sudo docker compose down && sudo docker compose up -d
    ```

### Troubleshooting

If you encounter a 503 error:
* Ensure you restarted Docker containers after updating `.env`.
* Confirm that the ALB's health check is configured with the correct path (`/api/v1/health`).
* Make sure the EC2 instance's security group allows traffic from the ALB.
* Look for more cues in the Nginx container logs:
    ```bash
    sudo docker compose logs nginx
    ```

For more troubleshooting guidance, see [this AWS troubleshooting guide](https://repost.aws/knowledge-center/alb-troubleshoot-503-errors).

> ## Accessing OpenOps via a custom domain over TLS

Reviewer: Sorry for misleading you, the custom domain mapping and TLS should stay together, and not be split into separate guides. I only meant that the title should reflect that.

Author: Reverted to the original version that only contains one section and clarified the title. Please check again

## TLS

For production usage, it’s recommended to enable TLS (HTTPS). You can achieve this by adding an EC2 Load Balancer to serve traffic to your EC2 instance, as shown in the next section on custom domains.

Without TLS, cloud templates may not load properly when using the Safari browser.

## Accessing OpenOps via a custom domain

If you want to access your OpenOps instance via a custom domain (e.g., `https://openops.your-company.com`) instead of its public IP address, you need to set up an AWS Application Load Balancer (ALB) with a valid TLS certificate from AWS Certificate Manager (ACM), then create a new DNS record and update your OpenOps configuration.

> 

Reviewer: The instructions are guiding the user to set up TLS (SSL) and not just a custom domain, it's worth pointing that out. If it was only about a custom domain, without TLS, a simple DNS record would've been enough, without ALB or a certificate.

Author: Thanks, good point. Considering that it may be useful for some to set up TLS without configuring a custom domain, I've split this into two separate procedures. Check out the latest updates.

## TLS

For production usage, it’s recommended to enable TLS (HTTPS). You can achieve this by adding an EC2 Load Balancer to serve traffic to your EC2 instance, as shown in the next section on custom domains.

In addition to the security aspect, enabling TLS also ensures that [workflow templates](/workflow-management/workflow-templates/) load properly in all browsers.

> ## Accessing OpenOps via a custom domain over TLS (HTTPS)
> 
> If you want to access your OpenOps instance via a custom domain (e.g., `https://openops.your-company.com`) instead of its public IP address, you need to set up an AWS Application Load Balancer (ALB) with a valid TLS certificate from AWS Certificate Manager (ACM), then create a new DNS record and update your OpenOps configuration.

Reviewer: Change the original title from "TLS" to "TLS and a Custom Domain". We can then remove the title here.

```suggestion
To setup a custom domain with TLS, we'll need to set up an AWS Application Load Balancer (ALB) with a valid certificate, generated by AWS Certificate Manager (ACM), create a new DNS record and then update the OpenOps configuration.
```
